src/components/Header.js

```diff
@@ -180,6 +182,10 @@ class Header extends Component {
     if (this.state.lastMsgTime !== this.props.message.time && this.props.message.text !== "") {
       this.setState({ snackOpen: true, lastMsgTime: this.props.message.time });
     }
+    if(shouldRender) {
```

Comment:
This doesn't look like it is really the way that we should be doing this. Something is weird.